napalm/base/helpers.py

```diff
@@ -258,52 +258,57 @@ def as_number(as_number_val):
         return int(as_number_str)
 
 
-def int_split_on_match(split_interface):
-    '''
-    simple fuction to split on first digit, slash, or space match
-    '''
-    head = split_interface.rstrip(r'/\0123456789 ')
-    tail = split_interface[len(head):].lstrip()
-    return head, tail
-
-
-def canonical_interface_name(interface, update_os_mapping=None):
-    '''
-    Function to retun interface canonical name
+def split_interface(intf_name):
+    """Split an interface name based on first digit, slash, or space match."""
+    head = intf_name.rstrip(r'/\0123456789. ')
+    tail = intf_name[len(head):].lstrip()
+    return (head, tail)
+
+
+def canonical_interface_name(interface, addl_name_map=None):
+    """Function to return an interface's canonical name (fully expanded name).
+
     This puposely does not use regex, or first X characters, to ensure
```

Comment:
You do understand this statement is completely false, don't you? Both using regex and `iface_name[0:2]` works perfectly fine in the example you are giving.

Reply:
My points with the previous comment are:

1. I want to make sure we are doing the right thing for the right reason. And that comment only makes me think the reason is not correct (maybe we are still doing the right thing though, not arguing that).
2. If the statement is false we should remove it.

Reply:
It will be a short time until you add case insensitivity to that statement which is what I was hinting at, but not explicitly stating. 

I cannot change your mind, but again, I have been normalizing interfaces for a while, and my regex attempts failed, and the cpan found &harmonizeIntsnhave not.

Reply:
> It will be a short time until you add case insensitivity to that statement which is what I was hinting at, but not explicitly stating.

What does this mean? regexes are case sensitive unless you specifically disable it

> I cannot change your mind, but again, I have been normalizing interfaces for a while, and my regex attempts failed, and the cpan found &harmonizeIntsnhave not.

It's not about changing my mind, I am not arguing about the implementation, I personally don't mind and I am not asking you to do it anyway. I just want to make sure we do not confusing people making false claims.

Reply:
Again, not trying to change the implementation, I just think we should remove that statement from the docstring: https://regex101.com/r/gAmu1N/1

Reply:
it’s in the doc string since it will come up again, over and over and it usually is easier to explain to future folks why not to do it to beat them to it with an explanation.

Reply:
I will make a more complete example